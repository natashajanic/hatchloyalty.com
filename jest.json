{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
