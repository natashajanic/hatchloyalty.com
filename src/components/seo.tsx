import { StaticQuery, graphql } from 'gatsby'
import * as React from 'react'
import Helmet from 'react-helmet'

type MetaEntry =
  | { name: string; content: any; property?: undefined }
  | { property: string; content: any; name?: undefined }

export interface IPureSEOProps {
  data: {
    site: {
      siteMetadata: {
        author: string
        description: string
        title: string
      }
    }
  }
  description?: string
  keywords?: string[]
  lang?: string
  meta?: MetaEntry[]
  title: string
}

export const PureSEO = (props: IPureSEOProps) => {
  const { data, description, keywords, lang, meta, title } = props
  const metaDescription = description || data.site.siteMetadata.description
  const metaEntries = [
    {
      name: 'description',
      content: metaDescription,
    },
    {
      property: 'og:title',
      content: title,
    },
    {
      property: 'og:description',
      content: metaDescription,
    },
    {
      property: 'og:type',
      content: 'website',
    },
    {
      name: 'twitter:card',
      content: 'summary',
    },
    {
      name: 'twitter:creator',
      content: data.site.siteMetadata.author,
    },
    {
      name: 'twitter:title',
      content: title,
    },
    {
      name: 'twitter:description',
      content: metaDescription,
    },
  ]
    .concat(
      keywords && keywords.length > 0
        ? {
            name: 'keywords',
            content: keywords.join(', '),
          }
        : []
    )
    .concat(meta || [])

  return (
    <Helmet
      htmlAttributes={{ lang: lang || 'en' }}
      title={title}
      titleTemplate={`%s | ${data.site.siteMetadata.title}`}
      meta={metaEntries}
    />
  )
}

interface ISEOProps {
  description?: string
  keywords?: string[]
  lang?: string
  meta?: MetaEntry[]
  title: string
}

class SEO extends React.Component<ISEOProps, {}> {
  render() {
    const query = graphql`
      query DefaultSEOQuery {
        site {
          siteMetadata {
            author
            description
            title
          }
        }
      }
    `
    return (
      <StaticQuery
        query={query}
        render={data => <PureSEO {...this.props} data={data} />}
      />
    )
  }
}

export default SEO
