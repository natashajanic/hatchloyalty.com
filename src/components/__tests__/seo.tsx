import { shallow, ShallowWrapper } from 'enzyme'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Helmet from 'react-helmet'
import { IPureSEOProps, PureSEO } from '../seo'

describe('Layout', () => {
  let props: IPureSEOProps
  let mountedComponent: ShallowWrapper<any, any, any>

  beforeEach(() => {
    props = {
      data: {
        site: {
          siteMetadata: {
            author: 'John Doe',
            description: 'Hatch Loyalty is modern customer activation.',
            title: 'Hatch Loyalty',
          },
        },
      },
      title: 'Hatch Loyalty',
    }
    mountedComponent = shallow(<PureSEO {...props} />)
  })

  describe('render', () => {
    it('renders a helmet', async () => {
      expect(mountedComponent.find(Helmet).length).toEqual(1)
      expect(
        mountedComponent
          .find(Helmet)
          .at(0)
          .prop('title')
      ).toEqual(props.title)
    })

    it('sets the language to english by default', async () => {
      expect(mountedComponent.find(Helmet).length).toEqual(1)
      expect(
        mountedComponent
          .find(Helmet)
          .at(0)
          .prop('htmlAttributes').lang
      ).toEqual('en')
    })

    it('sets the language to the specified prop', async () => {
      mountedComponent.setProps({ lang: 'de' })
      expect(mountedComponent.find(Helmet).length).toEqual(1)
      expect(
        mountedComponent
          .find(Helmet)
          .at(0)
          .prop('htmlAttributes').lang
      ).toEqual('de')
    })

    it('includes any specified meta entries', async () => {
      const testMeta = { name: 'test', content: 'Test Content' }
      mountedComponent.setProps({ meta: [testMeta] })
      expect(mountedComponent.find(Helmet).length).toEqual(1)
      expect(
        mountedComponent
          .find(Helmet)
          .at(0)
          .prop('meta')
      ).toContainEqual(testMeta)
    })

    it('includes the specified keywords as a meta entry', async () => {
      mountedComponent.setProps({ keywords: ['test', 'hatch'] })
      expect(mountedComponent.find(Helmet).length).toEqual(1)
      expect(
        mountedComponent
          .find(Helmet)
          .at(0)
          .prop('meta')
      ).toContainEqual({
        name: 'keywords',
        content: 'test, hatch',
      })
    })

    it('includes the specified description as a meta entry', async () => {
      mountedComponent.setProps({ description: 'Test Description' })
      expect(mountedComponent.find(Helmet).length).toEqual(1)
      expect(
        mountedComponent
          .find(Helmet)
          .at(0)
          .prop('meta')
      ).toContainEqual({
        name: 'description',
        content: 'Test Description',
      })
    })

    it('renders properly with overriden attributes', async () => {
      mountedComponent.setProps({
        description: 'Test Description',
        keywords: ['test', 'hatch'],
        lang: 'de',
        meta: [{ name: 'test', content: 'Test Content' }],
        title: 'Test Title',
      })
      expect(mountedComponent).toMatchSnapshot()
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
