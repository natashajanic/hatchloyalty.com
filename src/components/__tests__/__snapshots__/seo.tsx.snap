// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render matches the expected snapshot 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en",
    }
  }
  meta={
    Array [
      Object {
        "content": "Hatch Loyalty is modern customer activation.",
        "name": "description",
      },
      Object {
        "content": "Hatch Loyalty",
        "property": "og:title",
      },
      Object {
        "content": "Hatch Loyalty is modern customer activation.",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "John Doe",
        "name": "twitter:creator",
      },
      Object {
        "content": "Hatch Loyalty",
        "name": "twitter:title",
      },
      Object {
        "content": "Hatch Loyalty is modern customer activation.",
        "name": "twitter:description",
      },
    ]
  }
  title="Hatch Loyalty"
  titleTemplate="%s | Hatch Loyalty"
/>
`;

exports[`Layout render renders properly with overriden attributes 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "de",
    }
  }
  meta={
    Array [
      Object {
        "content": "Test Description",
        "name": "description",
      },
      Object {
        "content": "Test Title",
        "property": "og:title",
      },
      Object {
        "content": "Test Description",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "John Doe",
        "name": "twitter:creator",
      },
      Object {
        "content": "Test Title",
        "name": "twitter:title",
      },
      Object {
        "content": "Test Description",
        "name": "twitter:description",
      },
      Object {
        "content": "test, hatch",
        "name": "keywords",
      },
      Object {
        "content": "Test Content",
        "name": "test",
      },
    ]
  }
  title="Test Title"
  titleTemplate="%s | Hatch Loyalty"
/>
`;
